federation/pkg/federation-controller/sync/controller.go

```diff
@@ -384,6 +391,50 @@ func (s *FederationSyncController) delete(obj pkgruntime.Object, kind string, na
 	return nil
 }
 
+// updateFedSpec updates the spec against the federated object using the adapters UpdateFedSpecHook of need be.
+func (s *FederationSyncController) updateFedSpec(fedObj pkgruntime.Object, hook federatedtypes.UpdateFedSpecHook) reconcileStatus {
+	clusters, err := s.informer.GetReadyClusters()
```

Comment:
Agreed, and also as stated in previous comment, I think one additional method on the adapter also suffices. Am skipping this now, to first get an agreement into how we handle the needed additional functionality.
I have updated the hook functions to methods in a wrapper interface (taking suggestion from ur last comment)
@perotinus, need your comments too, if you think this is a better and simpler way of doing the whole stuff.